packages/sanity-plugin-nrkno-spellcheck/src/lib/core/document/document-spellcheck-values.ts

```diff
@@ -0,0 +1,150 @@
+  // the p.value type is not entirely correctly typed
+  // noinspection SuspiciousTypeOfGuard
+  return p.value && typeof p.value === 'string' && p.spellcheck;
+}
+
+function getPaths(object: any, documentType: ObjectSchemaType, language: Language) {
+  const paths: PathValue[] = [];
+  const nodeStack: PathValue[] = [
+    {
+      type: documentType,
+      value: object,
+      parentFieldTypes: [],
+      path: [] as Path,
+    },
+  ];
+
+  while (nodeStack.length > 0) {
+    const n = nodeStack.pop();
+    if (!n) {
+      break;
+    }
+
+    Object.entries(n.value)
+      .filter(([key]) => !key.startsWith('_'))
+      .forEach(([key, value]) => {
+        const parentType = n.type as ObjectSchemaType | ArraySchemaType;
+        let fieldType: SchemaType | undefined;
+        if (parentType.jsonType === 'object') {
+          fieldType = parentType.fields.find((f) => f.name === key)?.type;
+        } else {
+          fieldType = parentType.of.find((o) => o.type?.name === (value as any)._type);
+        }
+
+        const isWithinBlock = fieldType?.name === 'block' || !!n.blockParentType;
+
+        if (
+          !fieldType ||
+          isSpellcheckDisabledForPath(fieldType, language) ||
+          (isWithinBlock && ignoredBlockFields.includes(key))
+        ) {
+          return;
+        }
+
+        const keyedItem = (value as any)?._key;
+        const pathSegment: PathSegment = keyedItem ? { _key: keyedItem } : key;
+
+        const path = n.path.concat(pathSegment);
+        const blockParentType = fieldType.name === 'block' ? parentType : n.blockParentType;
+        const blockValue = fieldType.name === 'block' ? (value as unknown as Block) : n.blockValue;
+        const isSpanString = fieldType.name === 'string' && parentType.name === 'span';
+        let blockFieldPath;
+        if (isSpanString && path.length > 4) {
+          blockFieldPath = path.slice(0, path.length - 4);
+        }
+
+        const pathValue: PathValue = {
+          type: fieldType,
+          path,
+          parentFieldTypes: [...n.parentFieldTypes, n.type],
+          blockFieldPath,
+          blockParentType,
+          blockValue,
+          value,
+          spellcheck: canSpellcheckField(fieldType, language),
+        };
+        paths.push(pathValue);
+
+        // if value is object or array (this covers both), add to stack
+        if (typeof value === 'object') {
+          nodeStack.unshift(pathValue);
+        }
+      });
+  }
+  return paths;
+}
+
+function canSpellcheckField(fieldType: any, language: Language) {
+  const spellcheckableType = fieldType.name === 'text' || fieldType.name === 'string';
+
+  const options = fieldType?.options as DisablesSpellcheckOptions | undefined;
+  const disableSpellcheck = isSpellcheckDisabledForPath(fieldType, language) || !!options?.list;
+
+  return spellcheckableType && !disableSpellcheck;
+}
+
+function isSpellcheckDisabledForPath(fieldType: any, language: Language) {
```

Comment:
Her er det lagt til støtte for language-per-field og for å ignorere felter som har strengverdi men som vi ikke vil spellcheck (slug, reference ect)